airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py

```diff
@@ -146,10 +146,34 @@ def primary_keys_for_records(streams, records):
 @pytest.mark.default_timeout(5 * 60)
 class TestBasicRead(BaseTest):
     @staticmethod
-    def _validate_schema(records, configured_catalog):
+    def _validate_records_structure(records: List[AirbyteRecordMessage], configured_catalog: ConfiguredAirbyteCatalog):
+        """
+        Check object structure simmilar to one expected by schema. Sometimes
+        just running schema validation is not enough case schema could have
+        additionalProperties parameter set to true and no required fields
+        therefore any arbitrary object would pass schema validation.
+        This method is here to catch those cases by extracting all the pathes
+        from the object and compare it to pathes expected from jsonschema. If
+        there no common pathes then raise an alert.
+
+        :param records: List of airbyte record messages gathered from connector instances.
+        :param configured_catalog: SAT testcase parameters parsed from yaml file
+        """
+        schemas: Dict[str, Set] = {}
+        for stream in configured_catalog.streams:
+            schemas[stream.stream.name] = set(get_expected_schema_structure(stream.stream.json_schema))
+
+        for record in records:
+            record_fields = set(get_object_structure(record.data))
+            common_fields = set.intersection(record_fields, schemas.get(record.stream))
+            assert common_fields, f" Record from {record.stream} stream should have some fields mentioned by json schema"
```

Comment:
If I understand correctly, then at least one of the "properties" must match in the scheme and in the record for the entire nesting. I'm not sure if this is correct. There may be an object in the record, where we will not receive full nesting, but only partially, because there is simply no other data, and then this check will fall.

Reply:
Yes, in this case it would be different structure but we are not expected to match all the fields, one common field would be enough

Reply:
I think the only edge case here is if there an object with no declared properties and additionalProperties=true (which is technically valid) - is that handled here?